skimage/segmentation/chanvese.py

```diff
@@ -0,0 +1,175 @@
+    fxy = .25*(P[2:, 2:]+P[:-2, :-2]-P[:-2, 2:]-P[2:, :-2])
+    G = np.sqrt(fx**2+fy**2)
+    K = (fxx*fy**2-2*fxy*fx*fy+fyy*fx**2)/(np.power(fx**2+fy**2, 1.5)+1e-10)
+    KG = K*G
+    return KG/np.max(np.abs(KG))
+
+
+def _cv_heavyside(x, eps=1.):
+    """ Returns the result of a regularised heavyside function of the
+    input value(s).
+    """
+    return 0.5*(1.+2./np.pi*np.arctan(x/eps))
+
+
+def _cv_delta(x, eps=1.):
+    """ Returns the result of a regularised dirac function of the
+    input value(s).
+    """
+    return eps/(eps**2+x**2)
+
+
+def _cv_difference_from_average_term(U0, Hphi, lambda_pos, lambda_neg):
+    """ Returns the 'energy' contribution due to the difference from
+    the average value within a region at each point.
+    """
+    H = Hphi
+    Hinv = 1.-H
+    Hsum = np.sum(H)
+    Hinvsum = np.sum(Hinv)
+    c1 = np.sum(U0*H)
+    c2 = np.sum(U0*Hinv)
+    if(Hsum != 0):
+        c1 = c1/Hsum
+    if(Hinvsum != 0):
+        c2 = c2/Hinvsum
+    return lambda_pos*(U0-c1)**2*H+lambda_neg*(U0-c2)**2*Hinv
+
+
+def _cv_edge_length_term(phi, mu):
+    """ Returns the 'energy' contribution due to the length of the edge
+    between regions at each point, multiplied by a factor 'mu'.
+    """
+    toret = _cv_curvature(phi)
+    return mu*toret
+
+
+def _cv_energy(U0, phi, mu, lambda1, lambda2, heavyside=_cv_heavyside):
+    """ Returns the total 'energy' of the current level set function
+    """
+    H = heavyside(phi)
+    return np.sum(_cv_difference_from_average_term(U0, H, lambda1, lambda2) +
+                  _cv_edge_length_term(phi, mu))
+
+
+def _cv_signed_distance_from_zero_level(phi):
+    """ Returns a level set function corresponding to a signed distance
+    from the zero level of the input level set 'phi'.
+    """
+    return distance_transform_cdt(phi > 0)-distance_transform_cdt(phi <= 0)
+
+
+def _cv_initial_shape(U0):
+    """ Generates a checkerboard level set function. According to
+    Pascal Getreuer, such a level set function has fast convergence.
+    """
+    yv = np.arange(U0.shape[0]).reshape(U0.shape[0], 1)
+    xv = np.arange(U0.shape[1])
+    sizex = U0.shape[1]/10.
+    sizey = U0.shape[0]/10.
+    return (np.sin(np.pi/sizey*yv)*(np.sin(np.pi/sizex*xv)))
+
+
+def _cv_calculate_variation(U0, phi, mu, lambda1, lambda2,
+                            heavyside=_cv_heavyside):
+    """ Calculates the variation of the level set 'phi' function for
+    one step of the Chan-Vese algorithm. This value is not multipled by
+    the 'virtual time' expressed in the original article.
+    """
+    H = heavyside(phi)
+    Hinv = 1.-H
+    Hsum = abs(np.sum(H))
+    Hinvsum = abs(np.sum(Hinv))
+    c1 = np.sum(U0*H)
+    c2 = np.sum(U0*Hinv)
+    if(Hsum != 0):
+        c1 = c1/Hsum
+    if(Hinvsum != 0):
+        c2 = c2/Hinvsum
+    return _cv_edge_length_term(phi, mu)-lambda1*(U0-c1)**2+lambda2*(U0-c2)**2
+
+
+def chan_vese(U0, mu, lambda1=1., lambda2=1., tol=1., maxiter=100, dt=1):
```

Comment:
`skimage` convention is to use `img` as a name of input image variable. Please, rename here and in other functions.

Regarding default values for `lambda1, ...`, I think it is better to use `1.0` instead of `1.`. This looks better when function signature is being shown, and also in API docs.